integration/env/env.go

```diff
@@ -0,0 +1,74 @@
+package env
+
+import (
+	"context"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+)
+
+// Env should be created using Builder.
+type Env struct {
+	Azure   Azure
+	Common  Common
+	Cluster Cluster
+}
+
+type BuilderConfig struct {
+	Logger micrologger.Logger
+}
+
+type Builder struct {
+	logger micrologger.Logger
+}
+
+func NewBuilder(config BuilderConfig) (*Builder, error) {
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+
+	b := &Bilder{
+		logger: config.Logger,
+	}
+
+	return b, nil
+}
+
+func (b *Builder) Build(ctx context.Context) (Env, error) {
```

Comment:
We get context which can be used to call other APIs while constructing data types.

I also thought about `e2eSetup` where we cold provide `BuildAWS`, `BuildAzure` and `BuildKVM`. With with `init()` functions we'd either have a lot of code repetition or we'd have set e.g. set kvm variables in azure test. None of these approaches is good.